src/daft-local-execution/src/intermediate_ops/project.rs

```diff
@@ -12,15 +12,73 @@ use super::intermediate_op::{
 };
 use crate::NUM_CPUS;
 
+fn num_parallel_exprs(projection: &[ExprRef]) -> usize {
+    max(
+        projection
+            .iter()
+            .map(|expr| expr.has_compute())
+            .filter(|x| *x)
```

Comment:
tip: you can use `filter_map` here _(I'm surprised clippy doesnt recommend this)_